library/psa_crypto.c

```diff
@@ -785,8 +895,12 @@ static psa_status_t psa_get_and_lock_key_slot_with_policy(
         goto error;
 
     /* Enforce that the usage policy permits the requested algortihm. */
-    if( alg != 0 && ! psa_key_policy_permits( &slot->attr.policy, alg ) )
-        goto error;
+    if( alg != 0 )
+    {
+        status = psa_key_policy_permits( &slot->attr.policy, alg );
```

Comment:
With this change it would probably be better to move the `status = PSA_ERROR_NOT_PERMITTED;` above:
```C

    if( ( slot->attr.policy.usage & usage ) != usage )
    {
        status = PSA_ERROR_NOT_PERMITTED;
        goto error;
    }
```

Reply:
Done

Reply:
I disagree: I prefer to set `status` to a safe default (`NOT_PERMITTED`) and then explicitly authorize if warranted. This way, an omitted case in the implementation results in something not working (and likely to be caught by functional testing) rather than in a security hole. However, this function is so short that it doesn't matter, so please don't change it back. I'm just writing this comment for future reference.